src/states/Postroll.js

```diff
@@ -0,0 +1,139 @@
+import videojs from 'video.js';
+
+import {AdState, BeforePreroll, Preroll, AdsDone} from '../states.js';
+import {startAdBreak, endAdBreak} from '../adBreak.js';
+
+export default class Postroll extends AdState {
+
+  init(player) {
+    // Legacy name that now simply means "handling postrolls".
+    player.ads._contentEnding = true;
+
+    // Start postroll process.
+    if (!player.ads.nopostroll_) {
+      player.addClass('vjs-ad-loading');
+
+      this._postrollTimeout = player.setTimeout(function() {
+        player.trigger('adtimeout');
+      }, player.ads.settings.postrollTimeout);
+
+    // No postroll, ads are done
+    } else {
+      player.setTimeout(() => {
+        player.ads.debug('Triggered ended event (no postroll)');
+        this.contentResuming = true;
+        player.trigger('ended');
+      }, 1);
+    }
+  }
+
+  onAdsError(player) {
+    // In the future, we may not want to do this automatically.
+    // Integrations should be able to choose to continue the ad break
+    // if there was an error.
+    if (player.ads.inAdBreak()) {
+      player.ads.endLinearAdMode();
+    }
+  }
+
+  /*
+   * Start the postroll if it's not too late.
+   */
+  startLinearAdMode() {
+    const player = this.player;
+
+    if (!player.ads.isAdPlaying() && !this.contentResuming) {
```

Comment:
old name `isAdPlaying` should be changed, and use the method for content resuming too

Reply:
Updating